ci-bench/src/util.rs

```diff
@@ -35,8 +35,358 @@ impl KeyType {
     }
 }
 
+pub mod async_io {
+    //! Async IO building blocks required for sharing code between the instruction count and
+    //! wall-time benchmarks
+
+    use async_trait::async_trait;
+    use std::cell::{Cell, RefCell};
+    use std::collections::VecDeque;
+    use std::fs::File;
+    use std::future::Future;
+    use std::pin::{pin, Pin};
+    use std::rc::Rc;
+    use std::task::{Poll, RawWaker, RawWakerVTable, Waker};
+    use std::{io, ptr, task};
+
+    /// Block on a future that should complete in a single poll.
+    ///
+    /// Safe to use when the underlying futures are blocking (e.g. waiting for an IO operation to
+    /// complete, and returning Poll::Ready afterwards, without yielding in between).
+    ///
+    /// Useful when counting CPU instructions, because the server and the client side of the
+    /// connection run in two separate processes and communicate through stdio using blocking
+    /// operations.
+    pub fn block_on_single_poll(
+        future: impl Future<Output = anyhow::Result<()>>,
+    ) -> anyhow::Result<()> {
+        // We don't need a waker, because the future will complete in one go
+        let waker = noop_waker();
+        let mut ctx = task::Context::from_waker(&waker);
+
+        let future = pin!(future);
+        match future.poll(&mut ctx) {
+            Poll::Ready(result) => result,
+            Poll::Pending => {
+                panic!("future unexpectedly pending!")
```

Comment:
I'm a bit torn on this one... The panic here signals that the caller violated the function contract by providing a future that does not finish in one poll. However, since this is all internal code and we control the calling code, this match arm should indeed be unreachable. I'd rather keep the panic and use a more descriptive error message, like `the provided future did not finish after one poll`. What do you think?

Reply:
>  I'd rather keep the panic and use a more descriptive error message, like the provided future did not finish after one poll. What do you think?
 
That sounds good to me! I don't feel strongly about the suggestion.